.github/workflows/haskell-ci.yml

```diff
@@ -56,10 +56,9 @@ jobs:
     strategy:
       matrix:
         stack:
-          - 2.7.5
-        stack-yaml:
-          - stack-8.10.7.yaml
-          - stack-9.0.2.yaml
+          - 2.11.1
+        ghc:
```

Comment:
Is there a specific reason why you removed the old stack.yaml files? We've lost the guarantee that the library would work with older Stackage snapshots.

Reply:
@michivi I just wanted to have a single `stack` file. I meant to bring both cabal and stack to the same page by passing `ghc` field. As for guarantees with older version, I'd just add more versions to `ghc` field like @jhrcek suggests.

Reply:
I think it you want to combine using stack.yaml (which determines which stack-installed ghc will be used) and externally (by ghcup) installed ghc, you would have to use `stack build ... --system-ghc`.

But I find that unnecessarily complicated.

I'd prefer just having 2-3 stack yamls and passing those as matrix parameters + not specifying ghc versions as matrix parameters. That feels like more "stack native" way to do it.

Reply:
Yeah, if you're using a stackage LTS version, then you have an associated GHC version requirement. Using a different version than what is specified in your current resolver will most likely bring issue (unsatisfied constraints on `base`). Using the `matrix` parameters and different stack.yaml seems preferable.

Reply:
As an illustration that current approach doesn't work: your stack 9.0.2 job was actually using ghc 9.4.5

![Screenshot from 2023-06-29 07-48-54](https://github.com/Holmusk/postgresql-simple-named/assets/2716069/1a3b6f87-7054-4e83-8119-0ee1b46d6428)